### PR TITLE
Hardware event support

### PR DESCRIPTION
**Pull request by jzding**

### CODE REVIEW

File: routes.go [hunk 1]
```diff
@@ -302,6 +304,41 @@ func (s *Server) publishEvent(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+// publishEvent gets cloud native events and converts it to cloud event and publishes to a transport to send
```
Comment (aneeshkp) [inline routes.go:307]: the comment needs to be fixed related to hw event

File: routes.go [hunk 2]
```diff
@@ -302,6 +304,41 @@ func (s *Server) publishEvent(w http.ResponseWriter, r *http.Request) {
...
+	if err != nil {
+		localmetrics.UpdateEventPublishedCount(pub.Resource, localmetrics.FAIL, 1)
+		respondWithError(w, err.Error())
+	} else {
+		s.dataOut <- &channel.DataChan{
+			Type:    channel.EVENT,
+			Data:    ceEvent,
+			Address: pub.GetResource(),
+		}
+		localmetrics.UpdateEventPublishedCount(pub.Resource, localmetrics.SUCCESS, 1)
```
Comment (aneeshkp) [inline routes.go:337]: where are local metrics changes? pls run make test  and make lint

Comment (aneeshkp) [reply routes.go:337]: localmetrics.go

Comment (jzding) [reply routes.go:337]: make test and make lint all passed. I don't see the need to change localmetrics, unless we want to make the metrics name more generic: `cne_events_api_published` => `cloud_events_api_published`?

https://github.com/redhat-cne/rest-api/blob/main/pkg/localmetrics/localmetrics.go#L41

Comment (jzding) [reply routes.go:337]: make test and make lint all passed. I don't see the need to change localmetrics, unless we want to make the metrics name more generic: `cne_events_api_published` => `cloud_events_api_published`?

https://github.com/redhat-cne/rest-api/blob/main/pkg/localmetrics/localmetrics.go#L41

Comment (aneeshkp) [reply routes.go:337]: I am lookign for UpdateEventPublishedCount in localmetrics pakcage

**Review by aneeshkp — APPROVED** [review]

**Pull request merged**